aldryn_newsblog/cms_plugins.py

```diff
@@ -27,11 +31,41 @@ class NewsBlogPlugin(TemplatePrefixMixin, CMSPluginBase):
     module = 'News & Blog'
 
 
+class AdjustableCacheMixin(object):
+    """
+    For django CMS < 3.3.0 installations, we have no choice but to disable the
+    cache where there is time-sensitive information. However, in later CMS
+    versions, we can configure it with `get_cache_expiration()`.
+    """
+    if not CMS_GTE_330:
+        cache = False
+
+    def get_cache_expiration(self, request, instance, placeholder):
+        return getattr(instance, 'cache_duration', 0)
+
+    def get_fieldsets(self, request, obj=None):
+        """
+        Removes the cache_duration field from the displayed form if we're not
+        using django CMS v3.3.0 or later.
+        """
+        fieldsets = super(AdjustableCacheMixin, self).get_fieldsets(request, obj=None)
+        if CMS_GTE_330:
+            return fieldsets
+
+        field = 'cache_duration'
+        for fieldset in fieldsets:
+            if field in fieldset[1]['fields']:
+                new_fieldset = [
```

Comment:
You could as well remove the `if`, as the `in` operator causes the fields tuple to be iterated over anyway (so better once than twice).